src/blob_store.rs

```diff
@@ -127,13 +129,14 @@ impl<V: Blob> BlobStore<V> {
             base_path: path.clone(),
             _value_type: std::marker::PhantomData,
         };
+        // load pages
```

Comment:
Not strictly related, but a bit further up: we can probably preallocate the pages vector because we know `num_pages`.

Reply:
That's true, pushing